Sources/NIOHTTP1/HTTPServerPipelineHandler.swift

```diff
@@ -0,0 +1,227 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+//
+//  HTTPServerPipelineHandler.swift
+//  NIOHTTP1
+//
+//  Created by Cory Benfield on 01/03/2018.
+//
+
+import NIO
+
+/// A `ChannelHandler` that handles HTTP pipelining by buffering inbound data until a
+/// response has been sent.
+///
+/// This handler ensures that HTTP server pipelines only process one request at a time.
+/// This is the safest way for pipelining-unaware code to operate, as it ensures that
+/// mutation of any shared server state is not parallelised, and that responses are always
+/// sent for each request in turn. In almost all cases this is the behaviour that a
+/// pipeline will want. This is achieved without doing too much buffering by preventing
+/// the `Channel` from reading from the socket until a complete response is processed,
+/// ensuring that a malicious client is not capable of overwhelming a server by shoving
+/// an enormous amount of data down the `Channel` while a server is processing a
+/// slow response.
+///
+/// See https://tools.ietf.org/html/rfc7230#section-6.3.2 for more details on safely
```

Comment:
Consider using markdown syntax for the link: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links